sprints/3.md

```diff
@@ -0,0 +1,66 @@
+
+- Unit tests provide a better documentation how your class should behave. If you're a "normal programmer", you probably hate writing documentation. This forces you to write documentation while programming, and forces the documentation to never be obsolete (if it is, your tests fail). And it also helps when you need to change somebody else's code. - But one rarely needs a documentation for a single method.
+
+- https://stackoverflow.com/questions/67299/is-unit-testing-worth-the-effort?noredirect=1&lq=1
+
+- Unit Tests allows you to make big changes to code quickly. You know it works now because you've run the tests, when you make the changes you need to make, you need to get the tests working again. This saves hours. (bs cause you'll need to rewrite a lot)
+- TDD helps you to realise when to stop coding. Your tests give you confidence that you've done enough for now and can stop tweaking and move on to the next thing. (Especially when used with integration testing)
+- TDD helps with coding constipation - Especially true for integration testing
+- Unit Tests help you really understand the design of the code you are working on. - Not really, cause they mock the big picture
+- Unit Tests give you instant visual feedback, we all like the feeling of all those green lights when we've done. - Especially when using an integraion testing
+- Good unit tests can help document and define what something is supposed to do - And it's much more valuable to have a documentation for the app in general then to the individual models.
+- Unit tests help with code re-use. Purely non-sense.
+
+Tweets:
+- What is a ~woman~ unit?
+- The mystical ~unit~
+
+## A false sense of security
+
+Main point: Unit tests rarely covers the things that are the most likely to fail: dependency registration, external service calls, declarations like attributes.
+
+## Slower refactoring
+
+Unit testing frequently test internal components of your system. Example: validation put inside a model or a separate service. So when you decide to refactor those components you'll have to refactor your test + if you lack integration tests you won't be able to be sure if the code still works.
+
+## Fake documentation
+
+The documentation for a particular class is rarely needed and that's very complicated to understand what a software is doing because unit testing don't give you the bird-view picture you really need to understand a piece of software.
+
+## TDD has nothing to do with units
+
+All the benefits of TDD are much better exposed with integration testing:
+
+- Helps with coding constipation
+- Helps to realize when to stop coding.
+
+## What is unit vs integration
+
+May use "What is a ~woman~ unit?"
+
+Popular definition: Unit test tests the smallest unseparable component. Integration test tests the whole app behavior. The definition don't allow classify:
```

Comment:
A unit test covers the smallest inseparable component. While an integration test covers the whole app. The definition doesn't allow to classify